pkg/ast/spl/spl.peg

```diff
@@ -3295,10 +3453,92 @@ EarliestOnly <- CMD_EARLIEST EQUAL earliestTime:AbsoluteTimestamp !. {
     return node, nil
 }
 
-AbsoluteTimestamp <- [0-9][0-9] '/' [0-9][0-9] '/' [0-9][0-9][0-9][0-9] ':' [0-9][0-9] ':' [0-9][0-9] ':' [0-9][0-9] {
+RelIntegerAsString <- [-+][0-9]* {
+    return string(c.text), nil
+}
+
+WeekSnap <- ('w'[0-7]) {
     return string(c.text), nil
 }
 
+RelTimeUnit <- timeUnit:(Second / Month / Minute / Hour / Day / Week / Quarter / Year) {
+    return strconv.Itoa(int(timeUnit.(utils.TimeUnit))), nil
+}
+
+Snap <- '@' snap:(WeekSnap / RelTimeUnit) {
+    return snap.(string), nil
+}
+
+Offset <- off:(RelIntegerAsString) tuOff:(RelTimeUnit) {
+    var offsetNum int64 = 0
+    var err error
+    offStr := off.(string)
+    if offStr == "+" {
+        offsetNum = 1
+    } else if offStr == "-" {
+        offsetNum = -1
+    } else {
+        offsetNum, err = strconv.ParseInt(offStr, 10, 64)
+        if err != nil {
+            return nil, fmt.Errorf("Invalid Relative Time Format time unit value: (%v): , err: %v", offStr, err)
+        }
+    }
+
+    timeUnit, err := strconv.Atoi(tuOff.(string))
+    if err != nil {
+        return nil, fmt.Errorf("Invalid Time Unit value: (%v), err: %v", tuOff.(string), err)
+    }
+
+    relTimeOffset := ast.RelativeTimeOffset{
+        Offset: offsetNum,
+        TimeUnit: utils.TimeUnit(timeUnit),
+    }
+
+    return relTimeOffset, nil
+}
+
+RelativeTimestamp <- offset:(Offset)? snapParam:(Snap)? {
+    if offset == nil && snapParam == nil {
+        return nil, fmt.Errorf("Invalid Relative Time Format: Need either offset or snap param")
+    }
+
+    relTimeOffset := ast.RelativeTimeOffset{
+        Offset: 0,
+        TimeUnit: 0,
+    }
+    snap := ""
+    if snapParam != nil {
+        snap = snapParam.(string)
+    }
+
+    if offset != nil {
+        relTimeOffset.Offset = offset.(ast.RelativeTimeOffset).Offset
+        relTimeOffset.TimeUnit = offset.(ast.RelativeTimeOffset).TimeUnit
+    } 
+
+    relTimeModifier := ast.RelativeTimeModifier{
+        RelativeTimeOffset: relTimeOffset,
+        Snap: snap,
+        AbsoluteTime: "",
+    }
+
+    return relTimeModifier, nil
+}
+
+
+AbsoluteTimestamp <- [0-9][0-9] '/' [0-9][0-9] '/' [0-9][0-9][0-9][0-9] ':' [0-9][0-9] ':' [0-9][0-9] ':' [0-9][0-9] {
+    relTimeOffset := ast.RelativeTimeOffset{
+        Offset: 0,
+        TimeUnit: 0,
+    }
+    relTimeModifier := ast.RelativeTimeModifier{
+        RelativeTimeOffset: relTimeOffset,
+        Snap: "",
+        AbsoluteTime: string(c.text),
```

Comment:
You do not need to create default values like this. Go automatically creates the defaults. So you can directly do this:

```
relTimeModifier := ast.RelativeTimeModifier{
       AbsoluteTime: string(c.text)
}
// You can ignore any or all other fields, if you are going to assign some default values.
```

Reply:
Same comment for the above `RelativeTimestamp` definition.
You can just do this: `relTimeOffset := ast.RelativeTimeOffset`

Reply:
addressed in the latest commit.